src/main/scala/chiseltest/legacy/backends/verilator/VerilatorExecutive.scala

```diff
@@ -48,6 +55,115 @@ object VerilatorExecutive extends BackendExecutive {
     val circuit = elaboratedAnno.collect { case x: ChiselCircuitAnnotation => x }.head.circuit
     val dut = getTopModule(circuit).asInstanceOf[T]
 
+    if (elaboratedAnno.contains(CachingAnnotation)) {
+      val highFirrtlAnnos = (new ChiselStage).run(
+        elaboratedAnno :+ RunFirrtlTransformAnnotation(new HighFirrtlEmitter)
+      )
+
+      val rawFirrtl = highFirrtlAnnos.collect { case EmittedFirrtlCircuitAnnotation(e) => e.value }
+        .map(l => firrtl.Parser.parse(l))
+        .head
+
+      val sortedModuleHashStrings = new firrtl.stage.transforms.Compiler(targets = firrtl.stage.Forms.HighForm)
+        .transform(firrtl.CircuitState(rawFirrtl, Seq()))
+        .circuit
+        .modules
+        .map(m => m.name -> StructuralHash.sha256WithSignificantPortNames(m))
+        .sortBy(_._1)
+        .map(_._2.hashCode().toString) // TODO: Replace .hashCode().toString with .str
+
+      val sortedModuleMessageDigest = MessageDigest
+        .getInstance("SHA-256")
+        .digest(
+          sortedModuleHashStrings
+            .toString()
+            .getBytes("UTF-8")
+        )
+
+      val moduleHashHex = DatatypeConverter
+        .printHexBinary(sortedModuleMessageDigest)
+
+      val moduleHashFile = new File(targetDir, "module.hash")
+
+      val oldModuleHashHex = if (moduleHashFile.exists()) {
+        new BufferedReader(new FileReader(moduleHashFile)).readLine()
+      } else {
+        " "
+      }
+
+      val moduleWriter = new FileWriter(moduleHashFile)
+      moduleWriter.write(moduleHashHex)
+      moduleWriter.close()
+
+      // Hash the deterministic elements of elaboratedAnno for comparison, this should
+      // only catch if coverage or vcd flags change
+      val elaboratedAnnoString = elaboratedAnno.toSeq
+        .filter(anno =>
```

Comment:
could you replace this with something like?:

```.scala
.filter {
  case _ : chisel3.stage.ChiselCircuitAnnotatio | _ :  chisel3.stage.DesignAnnotation | _ : firrtl.options.TargetDirAnnotation => false
  case _ => true
}
```

Reply:
Done